src/overlay/Peer.cpp

```diff
@@ -1998,6 +1974,7 @@ Peer::recvFloodAdvert(StellarMessage const& msg)
         rememberHash(hash, seq);
     }
     mTxAdvertQueue.queueAndMaybeTrim(msg.floodAdvert().txHashes);
+    mApp.getOverlayManager().startDemandTimer();
```

Comment:
did this change because of the timer tests?

Reply:
The current pull-mode implementation starts the demand timer as soon as possible even if we haven't received an advert. This broke some timer tests as they're not aware of this new timer. The current workaround is to manually disable pull mode for those tests. 

Since this PR forces pull mode, the workaround is no longer possible. Instead of updating timer tests to accommodate pull mode, i decided to change the demand timer logic so it'll start demanding only after receiving the first advert.

Reply:
I don't think pull mode logic should be dictated by tests, as it's based on whatever policy we choose (in this case always start the timer), and the tests need to accommodate that. The problem you ran into isn't unique to pull mode, as there are other timers in the application. What you can do is cancel the demand timer, like we do with Herder timers here:
https://github.com/stellar/stellar-core/blob/master/src/util/test/TimerTests.cpp#L102